framework/encode/vulkan_state_tracker.cpp

```diff
@@ -3231,5 +3231,688 @@ void VulkanStateTracker::TrackBeginRendering(VkCommandBuffer commandBuffer, cons
     }
 }
 
+static void AssignDebugToObject(VkObjectType                        object_type,
+                                uint64_t                            object_handle,
+                                vulkan_state_info::CreateParameters object_name_parameter_buffer,
+                                vulkan_state_info::CreateParameters object_tag_parameter_buffer)
+{
+    switch (object_type)
+    {
+        case VK_OBJECT_TYPE_INSTANCE:
```

Comment:
This seems like a lot of identical code except for class names.  Will you tell me what you think about adding this to the generators?

Reply:
My initial approach was since the new members (`debug_name_create_parameters` and `debug_tag_create_parameters`) are in the base class to use polymorphism but failed to find a way because the base class is a template. 

I didn't look into autogenerating code for two reasons:
- The enumerates defined in `VkDebugReportObjectTypeEXT` are a superset of the objects we track so the autogen script will still need some logic to ignore those we don't handle.
- We already have hand written code with a switch statement per vulkan handle we handle in: [`framework/encode/vulkan_handle_wrapper_util.cpp`](https://github.com/LunarG/gfxreconstruct/blob/dev/framework/encode/vulkan_handle_wrapper_util.cpp)

Reply:
It sounds like this should be a later PR.

Just to be clear, you said "The enumerates defined in `VkDebugReportObjectTypeEXT`", but this function takes an `VkObjectType`.  Is that what it's supposed to take?

For the future, it looks like there are cases in this function for every object type except the video handles and the handles in `VK_NVX_binary_import`.  Since the registry specifies the VkObjectType for every handle, this seems like it could be generated.  But it looks like we don't already have generated code for state tracker, like you said.  So maybe vulkan_state_tracker and vulkan_state_writer and vulkan_handle_wrapper are just good candidates to hit with a code generator at some point.

Reply:
Yeah, it sounds like generating this should be a later PR.

Just to be clear, you said "The enumerates defined in `VkDebugReportObjectTypeEXT`", but this function takes an `VkObjectType`.  Is that what it's supposed to take?

For the future, it looks like there are cases in this function for every object type except the video handles and the handles in `VK_NVX_binary_import`.  Since the registry specifies the VkObjectType for every handle, this seems like it could be generated.  But it looks like we don't already have generated code for state tracker, like you said.  So maybe vulkan_state_tracker and vulkan_state_writer and vulkan_handle_wrapper are just good candidates to hit with a code generator at some point.